android/src/main/java/com/swmansion/gesturehandler/core/NativeViewGestureHandler.kt

```diff
@@ -127,6 +127,10 @@ class NativeViewGestureHandler : GestureHandler<NativeViewGestureHandler>() {
     this.hook = defaultHook
   }
 
+  fun markAsInBounds(value: Boolean) {
+    isWithinBounds = value
+  }
```

Comment:
Maybe changing it to something like this:
```suggestion
  fun withMarkedAsInBounds(closure: () -> Unit) {
    isWithinBounds = true
    closure()
    isWithinBounds = false
  }
```

(name tdb) and passing the state change chain in closure would be better? Then, this method could be moved to base `GestureHandler`, and `isWithinBounds` setter could stay private. Not sure though.

Reply:
Looks like a good idea, done in [5b6db76](https://github.com/software-mansion/react-native-gesture-handler/pull/2234/commits/5b6db7632b91de26c558aa7a169fac6b6b01639f)

Reply:
I've also tried to simplify a little `performClick` method. With this change I was able to navigate through the app using keyboard. Have a look at [8193cc8](https://github.com/software-mansion/react-native-gesture-handler/pull/2234/commits/8193cc8079d2df7a678b6242764aac449b78737d) and tell me what you think.